package-dev/Runtime/SentryInitialization.cs

```diff
@@ -106,6 +106,15 @@ public static void Init()
                     options.DiagnosticLogger?.LogDebug("Creating '{0}' span.", SubSystemSpanOperation);
                     SubSystemRegistrationSpan = InitSpan.StartChild(SubSystemSpanOperation, "subsystem registration");
                 }
+#endif
+            }
+            else
+            {
+                // Closing down the native layer that are set up during build and self-initialize
+#if SENTRY_NATIVE_COCOA
+                SentryNativeCocoa.Close(options.DiagnosticLogger);
```

Comment:
AFAICT [Cocoa native close](https://github.com/getsentry/sentry-cocoa/blob/fbeb49aa4bf63d1ce6537cbc4d509564d1510617/Sources/Sentry/SentrySDK.m#L405-L427) would crash here if sentry wasn't initialized. (Or maybe I understand objective-c wrong and it would first init the hub before closing it)

I think we cannot know here whether it was initialized - we could check options but that doesn't mean these are the same as the ones during build time. 

Therefore, we should check with the Cocoa SDK team & maybe update close() first to check whether hub is nil.